### ISSUE TITLE
adodpapi: issue with latest MSOLAP driver

### ISSUE THREAD

**Issue by LittleBigFrog** [mhammond/pywin32#1820]
Note that issues in this repository are only for bugs or feature requests in the pywin32.

**If you need support or help using this package, please follow [these instructions](https://github.com/mhammond/pywin32/blob/master/README.md#support)** - support or help requests will be closed without comment.

For all bugs, please provide the following information.

* Expected behavior and actual behavior.
conn = adodbapi.connect(ConnectionString) where ConnectionString starts with 
`Provider=MSOLAP.8;   Integrated Security=ClaimsToken;    Persist Security Info=True;`

Expected behavior: returns connection connection

Actual behavior: com_error: (-2147352567, 'Exception occurred.', (0, 'Microsoft OLE DB Provider for Analysis Services.', "Internal error: An unexpected error occurred (file 'pfadalwrapper.cpp', line 986, function 'PFAdalWrapper::GetInstance').", None, 0, -2147467259), None)
* Steps to reproduce the problem.
Install only MSOLAP driver avalable from [(https://docs.microsoft.com/en-us/analysis-services/client-libraries?view=asallproducts-allversions)] ( version 16)
Working fine with version 15.1.108.23 wich was available one month ago

* Version of Python and pywin32
Python 3.9.5 & pywin32 302
Python 3.9.1 &pywin32 228

**Comment by LittleBigFrog** [comment]
Think the issue is on new MSOLAP client library (v16), not pywin32

**Comment by vernondcole** [comment]
I was hoping that was the case. I have not had time to build a new VM for
testing yet. Please keep us updated.

**Comment by itsrobreally** [comment]
"A regression related to connections to cloud-based systems using AAD was discovered in the 16.0.4.17 version of OLEDB [MSOLAP]. It was fixed in the 16.0.9.17 version of MSOLAP."
https://docs.microsoft.com/en-us/analysis-services/client-libraries?view=asallproducts-allversions

**Comment by LittleBigFrog** [comment]
Issue still present with 16.0.9.17

**Comment by SanchK** [comment]
I had the same issue, so copied and pasted the entire "140" folder version of MSOLAP (15.1.108.3) from an older machine to my main machine, which had version 16. Is there a better method to install an older version of MSOLAP?

**Comment by yxchiwork** [comment]
I am still having the issue with newest MSOLAP version. Any solutions?

**Comment by itsrobreally** [comment]
The note on the MSOLAP page has changed, it now says 
"A regression related to connections to cloud-based systems using AAD was discovered in the 16.0.4.17 version of OLEDB (MSOLAP). It was fixed in the 16.0.9.17 version. Due to the nature of the issue, an installed 16.0.4.17 version cannot be corrected by simply upgrading the provider, even if setup is run in repair mode. It’s recommended to completely uninstall the 16.0.4.17 version and then install the 16.0.9.17 or later version."
I reverted to an old version of the driver until they get their act together, installing SSMS 18.10 gave me a good (but older) driver.
